evalml/pipelines/components/transformers/samplers/undersampler.py

```diff
@@ -169,7 +168,7 @@ def fit_resample(self, X, y):
             )
 
         y = infer_feature_types(y)
-
+        self.random_state = np.random.RandomState(self.random_seed)
```

Comment:
Moved this to make the undersampler deterministic!

Reply:
We don't need to store it, so I could replace this with 
```python
random_state = np.random.RandomState(...)
```
We use RandomState here to select a random subset of indexes to drop though! Although there are probably other methods of doing the same thing, I found this to be the most convenient method